### PR TITLE
Remark command

### PR DESCRIPTION
**Pull request by April0616**
Edits the remark for a person specified in the INDEX.
Format: remark INDEX r/[REMARK]

Examples:

remark 1 r/Likes to drink coffee.
Edits the remark for the first person to Likes to drink coffee.

remark 1 r/
Removes the remark for the first person.

### CODE REVIEW

**Review by nbriannl — DISMISSED** [review]
Failed Checkstyle. Please edit style.

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -747,6 +747,8 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |exit the program |
 
+|`* * *` |user |add a remark to a contact |see more details
```
Comment (nbriannl) [inline docs/DeveloperGuide.adoc:750]: straight forward user story 👍

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -171,6 +171,7 @@ Deletes the 2nd person in the address book.
 `delete 1` +
 Deletes the 1st person in the results of the `find` command.
 
+=======
```
Comment (nbriannl) [inline docs/UserGuide.adoc]: I will let this change pass. I will do another PR to patch formatting for everything in UserGuide.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * Edits the remark of a person to the address book.
+ */
+public class RemarkCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "remark";
+    public static final String COMMAND_ALIAS = "rm";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
```
Comment (nbriannl) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:26]: mention that the remark command also deletes the remark when you do a `remark r/`

Comment (nbriannl) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java:26]: `: If Remarks the person identified by the index number used in the last person listing.\n`
`  If the remark field is empty, the remark is removed for the person.`

Comment (nbriannl) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java:26]: ` "Example: (add remark)" + COMMAND_WORD + " 1" + PREFIX_REMARK + "Likes to drink coffee." `
` "Example: (delete remark)" + COMMAND_WORD + " 1" + PREFIX_REMARK" `

**Review by nbriannl — CHANGES_REQUESTED** [review]
Edit the command usage and everything is fine.
A remark is not during the add command, right? You only add a remark on an existing added person? Just to confirm how your feature works.

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java (hunk 3 above)
Comment (April0616) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java:26]: ok i'll check it

Comment (April0616) [reply src/main/java/seedu/address/logic/commands/RemarkCommand.java:26]: thanks

File: src/main/java/seedu/address/model/person/Person.java [hunk 4]
```diff
@@ -22,18 +22,19 @@
     private ObjectProperty<Phone> phone;
     private ObjectProperty<Email> email;
     private ObjectProperty<Address> address;
-
+    private ObjectProperty<Remark> remark;
     private ObjectProperty<UniqueTagList> tags;
 
     /**
-     * Every field must be present and not null.
+     * Constructor includes a remark field.
```
Comment (zacharytang) [inline src/main/java/seedu/address/model/person/Person.java]: Should change this back to the original comment, or improve on this comment. Currently it is missing the information from the previous comment

File: src/main/java/seedu/address/storage/XmlAdaptedPerson.java [hunk 5]
```diff
@@ -29,6 +30,8 @@
     private String email;
     @XmlElement(required = true)
     private String address;
+    @XmlElement(required = true)
```
Comment (zacharytang) [inline src/main/java/seedu/address/storage/XmlAdaptedPerson.java]: should this be required?

File: src/main/java/seedu/address/model/person/Person.java (hunk 4 above)
Comment (April0616) [reply src/main/java/seedu/address/model/person/Person.java]: oh i see,...at first I created two constructor....and finally i deleted one

File: src/main/java/seedu/address/storage/XmlAdaptedPerson.java (hunk 5 above)
Comment (April0616) [reply src/main/java/seedu/address/storage/XmlAdaptedPerson.java]: yeah, this field is required....
maybe that's why you met a problem adding a blindListener

File: src/main/java/seedu/address/logic/commands/RemarkCommand.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    private final Index targetIndex;
+    private final Remark remark;
+
+    /**
+     * @param targetIndex of the person in the list to edit the remark
+     * @param remark of the person
+     */
+    public RemarkCommand(Index targetIndex, Remark remark) {
+        requireNonNull(targetIndex);
```
Comment (zacharytang) [inline src/main/java/seedu/address/logic/commands/RemarkCommand.java:49]: could be replaced with requireAllNonNull

File: src/main/java/seedu/address/model/person/Person.java [hunk 7]
```diff
@@ -102,6 +103,18 @@ public Address getAddress() {
         return address.get();
     }
 
+    @Override
```
Comment (zacharytang) [inline src/main/java/seedu/address/model/person/Person.java:106]: May want to reorder the getters/setters to follow the ordering of current methods

**Review by zacharytang — APPROVED** [review]
Looks good!

**Review by nbriannl — APPROVED** [review]

**Pull request merged**